server/db/db.js

```diff
@@ -4,68 +4,114 @@
 
 'use strict';
 
+var crypto = require('crypto');
 var mysql = require('mysql');
+var uuid = require('uuid');
+
 var config = require('../config');
-var buf = require('buf').hex;
-var log = require('../logger')('server.db');
+var log = require('../logger')('server.db.db');
+var utils = require('./utils');
 
-var pool = mysql.createPool({
-  connectionLimit: config.get('db.mysql.connectionLimit'),
-  host: config.get('db.mysql.host'),
-  user: config.get('db.mysql.user'),
-  password: config.get('db.mysql.password'),
-  database: config.get('db.mysql.database')
-});
+var pool = utils.createPool();
 
-function _getConn(cb) {
-  pool.getConnection(function(err, conn) {
-    if (err) {
-      log.warn('error getting connection from pool: ' + err);
-      return cb(err);
-    }
-    cb(null, conn);
-  });
+function _trace(funcName, funcArgs) {
+  log.trace(funcName + ' called. arguments are: ' + [].join.call(funcArgs, ', '));
 }
 
 // DB API uses callbacks for the moment; TODO return promises?
+// TODO validate DB inputs from the API at some point
 module.exports = {
   createUser: function(fxaId, email, oauthToken, cb) {
-    var query = 'INSERT INTO users (fxa_id, email, oauth_token) ' +
-                'VALUES (?, ?, ?)' +
-                'ON DUPLICATE KEY UPDATE ' +
-                'fxa_id = VALUES(fxa_id), ' +
-                'email = VALUES(email), ' +
-                'oauth_token = VALUES(oauth_token)';
-    _getConn(function(err, conn) {
-      // TODO do we even want to handle pool connection errors here?
+    _trace('db.createUser', arguments);
+    var query = 'INSERT INTO users (fxaId, email, oauthToken) ' +
+                'VALUES (?, ?, ?)';
+    pool.query(query, [fxaId, email, oauthToken], function(err) {
```

Comment:
Re: `email VARCHAR(256) NOT NULL`,

Not 100% up to date on my MySQL skills, but should we be sanitizing the email address here and making sure that it isn't somehow magically longer than our 256 char field? I'm never sure if MySQL will truncate for us, or if we'll get an error.